### DISCUSSION TITLE
Running MSW in production

### DISCUSSION THREAD

**Discussion by alejandronanez** [mswjs/msw#1012] [Q&A] [answered]
Hey, I'll be working on a POC page that needs data from endpoints that are being developed for the next weeks or so. My page will be served on a _hidden_ url, and the endpoints will be brand new so there's not chance that I "override" an endpoint if I enable MSW in production (I believe)

My question here is, would it be safe/ok to enable MSW in prod for this particular case? 

**Possible downsides of this approach**
- Once the production endpoints are ready, it will be possible that the MSW still takes over and we can't send real data to the users.

Thanks for your help!

**Comment by kettanaito** [comment, marked as answer]
Hey, @alejandronanez.

You can use MSW in production for your POC. Endpoints you mock don't have to exist at all, MSW will perform request path matching regardless if the endpoint exists (that's the entire point!).

There are no downsides as to using the worker in production, except your users communicating with the mocked API. In the case of demos and POCs that's an affordable limitation, I think. 

> Once the production endpoints are ready, it will be possible that the MSW still takes over and we can't send real data to the users.

Once you enable the production endpoints, deploy the next version of your app where you don't call `worker.start()`. There's a self-destructive mechanism built-in into the library's worker that will unregister the pending worker if the client-side bundle doesn't call `worker.start()`. This minimizes the risk that your customers will continue communicating with the mock API after the release. 

> You may also utilize the fact that worker registrations are hostname-specific. If yo have a chance, consider deploying the POC with mocks on a different hostname (subdomains should work too).

**Reply by elFlaco1976** [reply]
Thanks, @kettanaito. So that means that this paragraph could be removed from [V1 docs](https://v1.mswjs.io/docs/getting-started/integrate/browser#start-worker)?

> It's not recommended to include Mock Service Worker in production. Doing so may lead to a distorted experience for your users.
